server.py

```diff
@@ -108,9 +108,20 @@ def remove_terms(content, repository_configuration, word_boundaries=True):
             content = re.compile(repo, re.IGNORECASE).sub("%s/repository/%s" % (self.public_url, repository_configuration["id"]), content)
             for term in repository_configuration['terms']:
                 if word_boundaries:
-                    content = re.compile(r'\b%s\b' % term, re.IGNORECASE).sub("XXX", content)
+                    regex = re.compile(r'\b%s\b' % term, re.IGNORECASE)
                 else:
-                    content = re.compile(term, re.IGNORECASE).sub("XXX", content)
+                    regex = re.compile(term, re.IGNORECASE)
+
+                if whole_urls:
+                    def sub_url(m):
+                        if regex.search(m.group(0)):
+                            return 'XXX'
+                        return m.group(0)
+
+                    url_regex = re.compile('\b((https?|ftp|file)://)[-A-Za-z0-9+&@#/%?=~_|!:,.;]+[-A-Za-z0-9+&@#/%=~_|]\b')
```

Comment:
```suggestion
                    url_regex = re.compile('((https?|ftp|file)://)[-A-Za-z0-9+&@#/%?=~_|!:,.;]+[-A-Za-z0-9+&@#/%=~_|]')
```

By changing this line, your code is able to anonymize the complete url

Reply:
Maybe it should be `\\b`?

Reply:
Indeed 👍